### ISSUE TITLE
training avg_dis maintaining around 0.024 on my own datasets, any advice?

### ISSUE THREAD

**Issue by YanqingWu** [j96w/DenseFusion#109]
I have checked the inputs, it's fine I am sure.
I only have one object for training.

**Comment by j96w** [comment]
Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.

**Comment by YanqingWu** [comment]
> Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.

do you know  this repo? https://github.com/F2Wang/ObjectDatasetTools.
I use it to make my dataset, I get about 500 images, may be more? its quality is ok, I think. Then I changed nothing but num objects. I am trying to train it with LINEMOD, to see if things will be better.

**Comment by marmas92** [comment]
I have the same issue - my avg_dis stays at around 0.018. Maybe it has something to do with the learning rate? I had to change my learning rate to 0.000001 so the avg_dis doesn't throw "nan" (as mentioned in issue https://github.com/j96w/DenseFusion/issues/105).

**Comment by j96w** [comment]
Hi @YanqingWu, I think 500 images with only a single object is a quite small dataset which might overfit easily. Could you take a try with more sampled data?

Hi @marmas92, have you turned on the training of Refinement model? The model I get on YCB dataset is about 0.013. Then I turned on the training of the refinement model to keep improving it.

**Comment by marmas92** [comment]
Thanks for the fast Feedback! 
I thougth the "default settings" for the training of Refinment model would be on "on". ^^ So should I set the refine_margin up to around 0.018?

**Comment by j96w** [comment]
`avg_dis` is in meters. Yes, you need to make sure that the model can trigger the `refine_margin` so that the training of the refinement model can start.

**Comment by marmas92** [comment]
Thnx for the help - now I get a pose_refine_model with avg_dis of 0.009 and a pose_model with an avg_dis of 0.013.

**Comment by YanqingWu** [comment]
> Hi @YanqingWu, I think 500 images with only a single object is a quite small dataset which might overfit easily. Could you take a try with more sampled data?
> 
> Hi @marmas92, have you turned on the training of Refinement model? The model I get on YCB dataset is about 0.013. Then I turned on the training of the refinement model to keep improving it.

@j96w I am trying ..., any progress I will let you know.

**Comment by j96w** [comment]
@marmas92 Sounds great.

**Comment by YanqingWu** [comment]
@j96w When I get more 1000 data, things get very good, everything is ok.

**Comment by j96w** [comment]
@YanqingWu Good to know.

**Comment by Ultraopxt** [comment]
> > Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.
> 
> do you know this repo? https://github.com/F2Wang/ObjectDatasetTools.
> I use it to make my dataset, I get about 500 images, may be more? its quality is ok, I think. Then I changed nothing but num objects. I am trying to train it with LINEMOD, to see if things will be better.

Hi Yanqingwu, sorry for disturbing you.
May I  ask how did you get gt.yml and  models_info.yml ? Cause I also use ObjectDatasetTools to make my dataset. I have got my dataset as following: 
rgb
depth
labels
transforms
centers.txt
farthest.txt
intrinsics.json
part.ply
radius.txt
transforms.npy
But without **gt.yml and  models_info.yml** .
Your any help will be really appreciated.

**Comment by YanqingWu** [comment]
> > > Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.
> > 
> > 
> > do you know this repo? https://github.com/F2Wang/ObjectDatasetTools.
> > I use it to make my dataset, I get about 500 images, may be more? its quality is ok, I think. Then I changed nothing but num objects. I am trying to train it with LINEMOD, to see if things will be better.
> 
> Hi Yanqingwu, sorry for disturbing you.
> May I ask how did you get gt.yml and models_info.yml ? Cause I also use ObjectDatasetTools to make my dataset. I have got my dataset as following:
> rgb
> depth
> labels
> transforms
> centers.txt
> farthest.txt
> intrinsics.json
> part.ply
> radius.txt
> transforms.npy
> But without **gt.yml and models_info.yml** .
> Your any help will be really appreciated.

Sorry, it has been too long ago. I don't remember  the details very well, but I am sure all you need is `ObjectDatasetTools` and `blender`, also a 3d camera like Kinetic. And you have to pay more attention in understanding the dataset.

**Comment by Ultraopxt** [comment]
> > > > Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.
> > > 
> > > 
> > > do you know this repo? https://github.com/F2Wang/ObjectDatasetTools.
> > > I use it to make my dataset, I get about 500 images, may be more? its quality is ok, I think. Then I changed nothing but num objects. I am trying to train it with LINEMOD, to see if things will be better.
> > 
> > 
> > Hi Yanqingwu, sorry for disturbing you.
> > May I ask how did you get gt.yml and models_info.yml ? Cause I also use ObjectDatasetTools to make my dataset. I have got my dataset as following:
> > rgb
> > depth
> > labels
> > transforms
> > centers.txt
> > farthest.txt
> > intrinsics.json
> > part.ply
> > radius.txt
> > transforms.npy
> > But without **gt.yml and models_info.yml** .
> > Your any help will be really appreciated.
> 
> Sorry, it has been too long ago. I don't remember the details very well, but I am sure all you need is `ObjectDatasetTools` and `blender`, also a 3d camera like Kinetic. And you have to pay more attention in understanding the dataset.

Thanks a lot.

**Comment by QRayMoor** [comment]
> > > > > Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.
> > > > 
> > > > 
> > > > do you know this repo? https://github.com/F2Wang/ObjectDatasetTools.
> > > > I use it to make my dataset, I get about 500 images, may be more? its quality is ok, I think. Then I changed nothing but num objects. I am trying to train it with LINEMOD, to see if things will be better.
> > > 
> > > 
> > > Hi Yanqingwu, sorry for disturbing you.
> > > May I ask how did you get gt.yml and models_info.yml ? Cause I also use ObjectDatasetTools to make my dataset. I have got my dataset as following:
> > > rgb
> > > depth
> > > labels
> > > transforms
> > > centers.txt
> > > farthest.txt
> > > intrinsics.json
> > > part.ply
> > > radius.txt
> > > transforms.npy
> > > But without **gt.yml and models_info.yml** .
> > > Your any help will be really appreciated.
> > 
> > 
> > Sorry, it has been too long ago. I don't remember the details very well, but I am sure all you need is `ObjectDatasetTools` and `blender`, also a 3d camera like Kinetic. And you have to pay more attention in understanding the dataset.
> 
> Thanks a lot.

@qingweihk 请问您的数据集做成功了吗？关于transforms.py 里的数据代表什么？RT加一行0001的齐次矩阵吗？请您指教

**Comment by JayLuoShui** [comment]
> 

怎么说，哥们？我也一开始用的ODT，然后卡在gt.yml,现在我倒是用bop_toolkit 搞定了models_info.yml,gt.yml到现在还没搞定捏

**Comment by JayLuoShui** [comment]
> > > > > > Hi, there is missing too much information for me to help. How is the quality of your own dataset? How many data do you have? How good are your ground truth labels? What parameters are you using to train? Also, training with a single object is easy to become overfitting when your generated pose changes are small.
> > > > > 
> > > > > 
> > > > > do you know this repo? https://github.com/F2Wang/ObjectDatasetTools.
> > > > > I use it to make my dataset, I get about 500 images, may be more? its quality is ok, I think. Then I changed nothing but num objects. I am trying to train it with LINEMOD, to see if things will be better.
> > > > 
> > > > 
> > > > Hi Yanqingwu, sorry for disturbing you.
> > > > May I ask how did you get gt.yml and models_info.yml ? Cause I also use ObjectDatasetTools to make my dataset. I have got my dataset as following:
> > > > rgb
> > > > depth
> > > > labels
> > > > transforms
> > > > centers.txt
> > > > farthest.txt
> > > > intrinsics.json
> > > > part.ply
> > > > radius.txt
> > > > transforms.npy
> > > > But without **gt.yml and models_info.yml** .
> > > > Your any help will be really appreciated.
> > > 
> > > 
> > > Sorry, it has been too long ago. I don't remember the details very well, but I am sure all you need is `ObjectDatasetTools` and `blender`, also a 3d camera like Kinetic. And you have to pay more attention in understanding the dataset.
> > 
> > 
> > Thanks a lot.
> 
> @qingweihk 请问您的数据集做成功了吗？关于transforms.py 里的数据代表什么？RT加一行0001的齐次矩阵吗？请您指教

怎么说，哥们？我也一开始用的ODT，然后卡在gt.yml,现在我倒是用bop_toolkit 搞定了models_info.yml,gt.yml到现在还没搞定捏
